tests/test_webhook.py

```diff
@@ -0,0 +1,224 @@
+# Copyright 2019 Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import absolute_import, division, print_function, unicode_literals
+
+import mock
+
+from c7n.actions.webhook import Webhook
+from .common import BaseTest
+
+
+class WebhookTest(BaseTest):
+
+    def test_valid_policy(self):
+        policy = {
+            "name": "webhook-batch",
+            "resource": "ec2",
+            "actions": [
+                {
+                    "type": "webhook",
+                    "url": "http://foo.com",
+                }
+            ],
+        }
+
+        self.assertTrue(self.load_policy(data=policy, validate=True))
+
+        policy = {
+            "name": "webhook-batch",
+            "resource": "ec2",
+            "actions": [
+                {
+                    "type": "webhook",
+                    "url": "http://foo.com",
+                    "batch": True,
+                    "parameters": {
+                        "foo": "bar"
+                    }
+                }
+            ],
+        }
+
+        self.assertTrue(self.load_policy(data=policy, validate=True))
+
+    @mock.patch('c7n.actions.webhook.requests.Request')
+    @mock.patch('c7n.actions.webhook.requests.Session')
+    def test_process_batch(self, session_mock, request_mock):
+        resources = [
```

Comment:
Since we are testing `batch` here, can we have multiple resources and do a parameter where we count the number of resources? Then we can assert that this parameter was actually calculated in the webhook.

Reply:
Updated to run in three batches and verify number of batches